test/helpers/value_hook.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:async';
+
+import 'package:test/test.dart' as test show expect, Matcher;
+
+/// Useful to easily test handler functions
+///  that set values outside of their scope.
+///
+/// ```dart
+/// // Construct VHook with initial value of 'false'
+/// VHook<bool> calledHandler = VHook<bool>(false);
+///
+/// willCallHandler((){
+///   calledHandler.set(true);
+///   // Some handler code
+/// });
+///
+/// // Test that the handler was called
+/// // This will throw an error if the handler was not called.
+/// calledHandler.expect(equals(true));
+/// ```
+class VHook<T> {
+  /// The value of the variable.
+  T? value;
+
+  /// Construct a new VariableHook with the given initial value.
+  VHook(T this.value);
```

Comment:
```suggestion
  T value;

  /// Construct a new VariableHook with the given initial value.
  VHook(this.value);
```

No need to make `value` an optional if you require a non optional value in the constructor

Reply:
Very much so actually. This would break the whole usecase of `VHook`. I often pass `null` to use the `.awaitValue(...)` method. Therefore value has to be an optional `T`.

Reply:
You don't it works as expected if you make value of type `T`. Here's why:

If `T` is non optional `value` will always be non optional because your constructor requires `T` not `T?`.
Therefore `value` is only null if you specify `T` as e.g. `String?`. This will continue to work.

```dart
// T is String? -> you can pass null.
final hook = VHook<String?>(null);

// T is String -> constructor will not accept null 
final hook = VHook<String>(null);
```